boa_engine/src/vm/code_block.rs

```diff
@@ -1034,10 +1054,21 @@ impl JsObject {
 
                 let (result, _) = result?;
 
-                if result.is_object() {
-                    Ok(result)
-                } else {
-                    Ok(frame.this.clone())
+                match result {
+                    JsValue::Object(ref obj) => Ok(obj.clone()),
+                    val => {
+                        if constructor.is_base() || val.is_undefined() {
+                            Ok(frame
+                                .this
+                                .as_object()
+                                .cloned()
+                                .expect("13. Assert: Type(thisBinding) is Object."))
```

Comment:
I'm guessing all this comes directly from the spec, right? Can we add comments from it as we do in other places?

Reply:
I added only that comment because it was appropriate for the expect. However, the definition of `construct` differs a lot from the spec, so it cannot be documented like every other function for now. I'll reanalyze `construct` to see if it can be rewritten in a way that conforms with the spec, so that we can document the whole function

Reply:
Sounds perfect to me :) thanks!!